projects/addon-editor/extensions/highlight/highlight.extension.ts

```diff
@@ -0,0 +1,40 @@
+import {getHTMLFromFragment} from '@tiptap/core';
+import {Highlight} from '@tiptap/extension-highlight';
+
+export const TuiHighlight = Highlight.extend({
+    addKeyboardShortcuts(): any {
+        return {
+            Tab: () => {
+                if (this.editor.isActive(`highlight`)) {
+                    const {selection, doc} = this.editor.state;
+                    const selected = doc.cut(selection.to, selection.to + 1);
+                    const sliced = getHTMLFromFragment(
+                        selected.content,
+                        this.editor.state.schema,
+                    ).replace(/<\/?[^>]+(>|$)/g, ``);
+                    const forwardSymbolIsWhitespace = sliced === ` `;
+
+                    if (forwardSymbolIsWhitespace) {
+                        this.editor.commands.setTextSelection(selection.to + 1);
+                    } else {
+                        this.editor.commands.setTextSelection(selection.to);
+
+                        this.editor.commands.insertContent(
+                            // require: `@tiptap/extension-text-style`
+                            `<span style="font-size: 16px"> </span>`,
+                        );
+                    }
+                }
+
+                return this.editor.commands.insertContent(`\t`);
+            },
+
+            Enter: () => this.editor.commands.insertContent(`<p></p>`),
+        };
+    },
+}).configure({
+    multicolor: true,
+    HTMLAttributes: {
+        class: `marked`,
```

Comment:
Do we need this class? Maybe people can just use `mark` tag name?

Reply:
fixed